docs/blueprints/EP025.rst

```diff
@@ -0,0 +1,111 @@
+:EP: 25
+:Title: Interface loop detection via LLDP
+:Authors:
+    - Italo Valcy <idasilva AT fiu DOT edu>
+    - Vinicius Arcanjo <vindasil AT fiu DOT edu>
+    - Antonio Francisco <ajoaoff AT gmail DOT com>
+    - Jeronimo Bezerra <jbezerra AT fiu DOT edu>
+    - Rogerio Motitsuki <rogerio.motitsuki AT gmail DOT com>
+:Created: 2022-02-22
+:Kytos-Version:
+:Status: Draft
+
+*****************************************
+EP025 - Interface loop detection via LLDP
+*****************************************
+
+
+Abstract
+========
+
+This blueprint proposes a loop detection mechanism on interfaces of a switch via LLDP, leveraging ``of_lldp`` existing capabilities, and also which events, states and configuration options will be available.
+
+
+Motivation
+==========
+
+A loop on a switch interface can be intentionally or not. Typically, intentional loops are for re-injecting packets in the switch pipeline to perform another packet transformation. On the other hand, unintentional loops can lead to major dataplane problems, especially if that interface has significant network traffic being being forwarded. A loop detection mechanism via LLDP will empower operators to define certain actions to be taken when a loop happens.
+
+
+Specification
+=============
+
+It is desirable to support ignored loops configuration that can be changed dynamically via API. Since this feature is relatively small and wouldn't require much extra entries or states, then using `topology` switches' and interfaces' metadata storage would simplify the implementation and maintenance since currently adding support to a new storage with storehouse requires a significant effort development-wise. In the future, this can be migrated to another storage (collection) when an alternative is available. The following metadata would be used:
+
+- ``switches.<dpid>.metadata.ignored_loops``:
+
+This is for configuring which loops should be ignored on a ``<dpid>``, the value is a list of list pair of interface port numbers. For instance, if it's desirable to ignore a loop between interface number 5 and 6, and also between port 1 and itself: ``[[5, 6], [1,1]]``
+
+- ``interfaces.<interface_id>.metadata.looped``:
+
+This ``looped`` attribute is for state management, it will contain the following object whenever a loop is in place, the ``detected_at`` string datetime will be in UTC without any ``Z`` modifier:
```

Comment:
`detected_at` will be the time when the loop was first seen or last seen? It would be useful for operators to know when the loop was first detected, while it would be helpful for the cleanup mechanism to know when it was last seen.

Reply:
`detected_at` exposed on `metadata.looped` currently represents when it was first seen, and then we also have a `updated_at`, that is an internal implementation (not currently exposed on metadata), which the clean up mechanism relies on it to confirm when it was last seen. yep, so it's aligned with this comment you've made, Italo. I'll also make an update in the blueprint trying to explain it better just so future readers don't get confused.

Reply:
Updated it on this commit https://github.com/kytos-ng/kytos/pull/181/commits/8f0209873300ceea6ec05a74b34b7dde76356131, let me know if you think it's clearer.

Reply:
Very nice, Vinicius! Yes it is very clear, and having the `updated_at` an internal implementation will be very helpful in the future!